### PR TITLE
Updated dmi drivers according to standalone

### PR DESCRIPTION
**Pull request by TillRasmussen**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    Cleanup, allowed hycom_atm_data  to have leap year and update dmi driver to reflect changes in standalone driver
- [x] Developer(s): 
tar
- [x] Suggest PR reviewers from list in the column to the right. Tony+Elizabeth
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
Limited differences in standalone code. This changes dmi driver according to updated standalone driver
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [x] No Unless the addition of leap years to hycom_atm_data should be documented
- [x] Please provide any additional information or relevant details below:
This should not change results. I have removed comments from ice_grid related to the calculation of anglet. Mostly added keywords to dmi driver as required by the updated icepack. At last allowed for hycom atm data to use leap years.

### CODE REVIEW

File: cicecore/drivers/nuopc/dmi/CICE_RunMod.F90 [hunk 1]
```diff
@@ -39,12 +39,12 @@ module CICE_RunMod
 !
 !  author Elizabeth C. Hunke, LANL
 !         Philip W. Jones, LANL
-!         William H. Lipscomb, LANL
+         William H. Lipscomb, LANL
```
Comment (apcraig) [inline cicecore/drivers/nuopc/dmi/CICE_RunMod.F90]: This looks like an error.  You probably need ! in front.

**Comment by apcraig** [comment]
travis is not passing.  I believe it cannot find the Icepack submodule version or change that's in this PR.  That should be fixed and reset to the current CICE Icepack version, 68bc397.  If you need some help with that, let me know.

File: cicecore/drivers/nuopc/dmi/CICE_RunMod.F90 (hunk 1 above)
Comment (TillRasmussen) [reply cicecore/drivers/nuopc/dmi/CICE_RunMod.F90]: Yes your are right.

**Comment by TillRasmussen** [comment]
Yes I needed to add "!"
lal.dat was a file that contained some diffs. (git diff ....). This has been removed
Icepack was the last (hopefully) leftover of my fork being corrupted when linked to icepack. 
 Pull should be updated. Check is ongoing

**Comment by phil-blain** [comment]
@TillRasmussen if you (re)take a look at https://github.com/CICE-Consortium/About-Us/wiki/Git-Workflow-Guidance maybe it will clarify things for you regarding how to deal with icepack. I added more details to that page. Essentially each time you switch branches with `git checkout` you have to follow with `git submodule update` for the Icepack submodule to stay in sync with the current branch. Same thing when you `git pull`, you have to follow with  `git submodule update`.

**Comment by TillRasmussen** [comment]
thanks, I did manage to get through.

Till

**Pull request merged**